File: BudgetApp.Client/src/app/entities/IncomeSource.ts
export class IncomeSource {
  incomeSourceId: string;
  name: string;
  amount: number;
  sortOrder: number;

  constructor(incomeSourceId: string, name: string, amount: number, sortOrder: number) {
    this.incomeSourceId = incomeSourceId;
    this.name = name;
    this.amount = amount;
    this.sortOrder = sortOrder;
  }
}

File: BudgetApp.Domain/Entities/IncomeSource.cs
namespace BudgetApp.Domain.Entities
{
    public class IncomeSource
    {
        public Guid IncomeSourceId { get; set; }
        public string Name { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public int SortOrder { get; set; }
    }
}